### PR TITLE
[DES-260] Blockly - Delete Function Modal Delete Button Color Issues

### PR DESCRIPTION
**Pull request by levadadenys**
### [DES-260] Blockly - Delete Function Modal Delete Button Color Issues
 * add dangerCancel Button logic for dialogButtons and LegacyButton
 
 Only works with this PR https://github.com/code-dot-org/blockly/pull/317 and blockly version bump
 
The main difficulty here is that we've had some wrong styles and parameters set in Blockly, however they won't work as expected without the changes made in this CDO PR cause this PR along with bumping Blockly version creates a fix for that edge case that we initially had to update in blockly.
 
And some more info for better understanding: 
The main problem here is that we already have a set of dialogs and button defined in blockly. While doing Phase 1.5 I was simply updating the cdo styles expecting all the dialogues to be consistent. But it turned out that they are not as consistent and not always used correct naming/structure/styles, just like the button I'm working on in this pr. It was a cancel button that was painted in red with the help of injected css with !important.

So, in short, the idea here is to fix and simplify where possible what we have right now without going too deep and rewriting too much of cdo-blockly (keeping in mind that we're going to get rid of it at some place)
 
 
 
Before:
image

After:
image
 
## Links
More info here: https://github.com/code-dot-org/blockly/pull/317
Jira [DES-260](https://codedotorg.atlassian.net/browse/DES-260?atlOrigin=eyJpIjoiYzNiOWMyY2ZlYTE1NDg2OWJmYWFmNmU3OTU2NjAwZTgiLCJwIjoiaiJ9)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/feedback.js [hunk 1]
```diff
@@ -1327,7 +1328,7 @@ FeedbackUtils.prototype.showSimpleDialog = function(options) {
       <DialogButtons
         confirmText={options.confirmText}
         cancelText={options.cancelText}
-        cancelButtonClass={options.cancelButtonClass}
+        isDangerCancel={!!options.isDangerCancel}
```
Comment (mikeharv) [inline apps/src/feedback.js:1331]: Can you explain why this is was called a "cancel" button? Usually, I would expect that a cancel button goes back without making any changes. Isn't that the "Keep" button in this case? Or does "cancel" just  mean it's the secondary/non-default button?

**Review by mikeharv — COMMENTED** [review]
Do these changes work without having to edit anything on the cdo Blockly repo?

**Comment by levadadenys** [comment]
> Do these changes work without having to edit anything on the cdo Blockly repo?

nope. We need to update blockly, then update cdo

File: apps/src/feedback.js (hunk 1 above)
Comment (levadadenys) [reply apps/src/feedback.js:1331]: well, that's a good question :D 
The main problem here is that we already have a set of dialogs and button defined in blockly. While doing Phase 1.5 I was simply updating the cdo styles expecting all the dialogues to be consistent. But it turned out that they are not as consistent and not always used correct naming/structure/styles, just like the button I'm working on in this pr. It was a cancel button that was painted in `red` with the help of injected css with `!important`.

So, in short, the idea here is to fix and simplify where possible what we have right now without going too deep and rewriting too much of cdo-blockly (keeping in mind that we're going to get rid of it at some place)

**Review by mikeharv — APPROVED** [review]
Approved. I feel good about merging this once it includes the blockly version bump. Feel free to ping me when that is ready.
